x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/field_parameters/analyzers_parameter.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { EuiSpacer } from '@elastic/eui';
+
+import { UseField, CheckBoxField, FormDataProvider } from '../../../shared_imports';
+import { NormalizedField } from '../../../types';
+import { getFieldConfig } from '../../../lib';
+import { EditFieldSection } from '../fields/edit_field';
+import { AnalyzerParameter } from './analyzer_parameter';
+
+interface Props {
+  field: NormalizedField;
+  withSearchQuoteAnalyzer?: boolean;
+}
+
+export const AnalyzersParameter = ({ field, withSearchQuoteAnalyzer = false }: Props) => {
+  return (
+    <EditFieldSection
+      title={i18n.translate('xpack.idxMgmt.mappingsEditor.analyzersSectionTitle', {
+        defaultMessage: 'Analyzers',
+      })}
+    >
+      <FormDataProvider pathsToWatch="useSameAnalyzerForSearch">
+        {({ useSameAnalyzerForSearch }) => {
+          const label = useSameAnalyzerForSearch
+            ? i18n.translate('xpack.idxMgmt.mappingsEditor.indexSearchAnalyzerFieldLabel', {
+                defaultMessage: 'Index and search analyzer',
+              })
+            : i18n.translate('xpack.idxMgmt.mappingsEditor.indexAnalyzerFieldLabel', {
+                defaultMessage: 'Index analyzer',
+              });
+
+          return (
+            <AnalyzerParameter path="analyzer" label={label} defaultValue={field.source.analyzer} />
+          );
+        }}
+      </FormDataProvider>
+
+      <EuiSpacer size="s" />
+
+      <UseField
+        path="useSameAnalyzerForSearch"
+        component={CheckBoxField}
+        config={{
+          label: i18n.translate(
+            'xpack.idxMgmt.mappingsEditor.analyzers.useSameAnalyzerIndexAnSearch',
+            {
+              defaultMessage: 'Use the same analyzers for index and searching',
+            }
+          ),
+          defaultValue: true,
```

Comment:
@cjcenizal This is what I tried to avoid, defining default values in multiple places in the code. I'd prefer to have a single source of truth for the parameter definition (their label, default value, ...). There is a helper function that we use for that: `getFieldConfig(<parameter>)`